scipy/sparse/construct.py

```diff
@@ -715,10 +715,8 @@ def random(m, n, density=0.01, format='coo', dtype=None,
     --------
     >>> from scipy.sparse import random
     >>> from scipy import stats
-    >>> class CustomRandomState(object):
-    ...     def randint(self, k):
-    ...         i = np.random.randint(k)
-    ...         return i - i % 2
+    >>> class CustomRandomState(int):
```

Comment:
Perhaps not ideal since this custom class is not so instructive, but this did help get the doctest working. I suspect feeding in an `int`-like object is just an easy way to avoid the change in infrastructure affected by this PR.

Reply:
This doesn't really make sense anymore now, `CustomRandomState` should be doing something. I'll have a look now. Better to skip the doctest rather than on purpose make an example worse.

Reply:
okay gave that another go, making `CustomRandomState` a `RandomState` subclass. It's still a weird thing, we could also remove it. But tests should pass now.